File: backend-spring-boot/src/main/java/no/vicx/backend/user/validation/CurrentPassword.java
package no.vicx.backend.user.validation;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Constraint(validatedBy = CurrentPasswordValidator.class)
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface CurrentPassword {
    String message() default "{vicx.constraints.CurrentPassword.message}";
    int minLength() default 1;
    int maxLength() default 1000;
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: backend-spring-boot/src/main/kotlin/no/vicx/backend/user/validation/CurrentPassword.kt
package no.vicx.backend.user.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass


@Constraint(validatedBy = [CurrentPasswordValidator::class])
@Target(AnnotationTarget.FIELD)
@Retention(
    AnnotationRetention.RUNTIME
)
annotation class CurrentPassword(
    val message: String = "{vicx.constraints.CurrentPassword.message}",
    val minLength: Int = 1,
    val maxLength: Int = 1000,
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)
